common/log.h

```diff
@@ -0,0 +1,658 @@
+        {                                                                                                                                    \
+            fprintf(LOG_TEE_TARGET, LOG_TEE_TIMESTAMP_FMT LOG_TEE_FLF_FMT str "%s" LOG_TEE_TIMESTAMP_VAL LOG_TEE_FLF_VAL "", ##__VA_ARGS__); \
+            fflush(LOG_TEE_TARGET);                                                                                                          \
+        }                                                                                                                                    \
+    }
+#endif
+
+// The '\0' as a last argument, is a trick to bypass the silly
+//  "warning: ISO C++11 requires at least one argument for the "..." in a variadic macro"
+//  so we can have a single macro which can be called just like printf.
+
+// Main LOG macro.
+//  behaves like printf, and supports arguments the exact same way.
+//
+#ifndef _WIN32
+/**/#define LOG(...) _LOG(__VA_ARGS__, "")
+#else
+/**/#define LOG(str, ...) _LOG("%s" str, "", __VA_ARGS__, "")
+#endif
+
+// Main TEE macro.
+//  does the same as LOG
+//  and
+//  simultaneously writes stderr.
+//
+// Secondary target can be changed just like LOG_TARGET
+//  by defining LOG_TEE_TARGET
+//
+#ifndef _WIN32
+/**/#define LOG_TEE(...) _LOG_TEE(__VA_ARGS__, "")
+#else
+/**/#define LOG_TEE(str, ...) _LOG_TEE("%s" str, "", __VA_ARGS__, "")
+#endif
+
+// LOG macro variants with auto endline.
+#ifndef _WIN32
+/**/#define LOGLN(...) _LOG(__VA_ARGS__, "\n")
+/**/#define LOG_TEELN(...) _LOG_TEE(__VA_ARGS__, "\n")
+#else
+/**/#define LOGLN(str, ...) _LOG("%s" str, "", __VA_ARGS__, "\n")
+/**/#define LOG_TEELN(str, ...) _LOG_TEE("%s" str, "", __VA_ARGS__, "\n")
+#endif
+
+// INTERNAL, DO NOT USE
+inline FILE *_log_handler1(bool change = false, LogTriState disable = LogTriStateSame, std::string filename = LOG_DEFAULT_FILE_NAME, FILE *target = nullptr)
+{
+    static bool _initialized{false};
+    static bool _disabled{(filename.empty() && target == nullptr)};
+    static std::string log_current_filename{filename};
+    static FILE *log_current_target{target};
+    static FILE *logfile = nullptr;
+
+    if (change)
+#ifndef _WIN32
+        [[unlikely]]
+#endif
+    {
+        if (disable == LogTriStateTrue)
+        {
+            // Disable primary target
+            _disabled = true;
+        }
+        // If previously disabled, only enable, and keep previous target
+        else if (disable == LogTriStateFalse)
+        {
+            _disabled = false;
+        }
+        // Otherwise, process the arguments
+        else
+        {
+            if (log_current_filename.compare(filename) != 0)
+            {
+                _initialized = false;
+            }
+
+            if (log_current_target != target)
+            {
+                _initialized = false;
+            }
+        }
+    }
```

Comment:
The above chain of conditionals could be made a lot shorter. At least use a boolean `||` for the last two checks: `log_current_filename == filename || log_current_target != target` (I see no reason to use 'compare' here).
What is our code style w/r/t braces? I'd prefer omitting them in the one-line blocks.